test/xbgpu/test_bsend.py

```diff
@@ -197,32 +212,58 @@ async def test_send_simple(
         outputs, time_converter, sensors
             Fixtures.
         """
+        # We need the "total number of engines required" to process this array
+        # configuration.
+        n_engines = 1
+        while n_engines < num_ants * 4:
+            n_engines *= 2
+
+        # The test still needs to have some idea of "which engine" this is in a
+        # sequence of B-engines.
+        engine_id = 4  # Arbitrarily chosen
```

Comment:
I don't think it's necessary to repeat this test with different numbers of antennas, since it's just a very indirect way to control the number of engines. My preference would be just to test with a fixed number of engines (say 8), but you could also put n_engines into the pytest.mark.combinations with a few values.

Setting engine_id to 4 when n_engines might be <= 4 is also invalid.

Reply:
Hmmmm, I naively updated the line to read `engine_id = 4 if n_engines > 4 else 3`, but that seems ~unnecessary.

Easier to set it to 3 which satisfies all variant of `num_ants`, and update the comment to explain it's not completely arbitrarily chosen.